docs/modules/storage/pages/recover-single-member.adoc

```diff
@@ -1,52 +1,88 @@
+|Number of seconds that the cluster allows for members to finish restoring data from their local persistence store. See the xref:configuring-persistence.adoc#persistence-data-load-timeout-seconds[`data-load-timeout-seconds`] configuration option.
+
+|master member
+|The oldest member in a cluster.
 
-== How Clusters Handle Single-Member Failures
+|partition table
+|Table that stores members' partition replica assignments and the partition table version.
+
+|persistence store
+|The files that contain persisted data.
+
+|validation timeout
+|Number of seconds that the cluster allows for members to rejoin and send their partition table to the master member. See the xref:configuring-persistence.adoc#persistence-validation-timeout-seconds[`validation-timeout-seconds`] configuration option
+|===
 
-By default, if a cluster detects that a member is restarting and attempting to rejoin, the cluster’s master member will ask the member to send its partition table for validation.
+== What Happens When a Single Member Leaves the Cluster
 
-If the master member validates that the partition table was correct at the time the member left the cluster, the member loads its persisted data from disk.
+By default, if a cluster detects that a missing member is restarting and attempting to rejoin, the cluster’s master member will ask the rejoining member to send its partition table for validation.
 
-If the cluster cannot validate the member's partition table for whatever reason, the member stops trying to join the cluster and shuts down. The cluster will not allow the member to rejoin until it has deleted its persistence store. The cluster then recovers the missing member's data from backups and
-redistributes it among the remaining cluster members.
+The rejoining member loads its persisted data from disk and rejoins the cluster only if the following conditions are met:
 
-You can configure what should happen if a cluster cannot validate a member's partition table. See <<handling-invalid-partitions, Handling Invalid Partitions>> on this page.
+- The master member receives the partition table within the validation timeout.
+- The master member validates that the partition table was correct at the time that the rejoining member left the cluster.
+- The rejoining member loads its persisted data from disk within the data load timeout.
 
-While a cluster is migrating a member's partitions, its performance may be affected by reduced throughput or temporary unavailability of data. If you have lots of persisted data and you are concerned about how long it may take for your cluster to migrate data after a member fails to rejoin, you can configure the following options:
+If these conditions are not met, the rejoining member stops trying to join the cluster and shuts down. The cluster does not allow that member to rejoin until it has deleted its persistence store. The cluster then tries to recover the missing member’s data from backups and repartitions the data.
```

Comment:
Should we perhaps describe first the default behaviour (by default `autoRemoveStaleData` is `true`)? So instead of `the rejoining member stops trying to join the cluster and shuts down` -> `the rejoining member automatically deletes its persistence store and rejoins the cluster`. And then we can add the passage that describes what happens when `autoRemoveStaleData` is `false` (member shuts down).